src/indexing.rs

```diff
@@ -191,24 +198,24 @@ pub(crate) mod tests {
     #[derive(Clone, Debug)]
     pub(crate) struct TestIndexingScheme;
 
-    impl IndexingScheme<(), usize> for TestIndexingScheme {
-        type Key = usize;
+    impl IndexingScheme<()> for TestIndexingScheme {
+        type Map = HashMap<usize, usize>;
 
         fn valid_bindings(
             &self,
-            key: &Self::Key,
-            known_bindings: &impl IndexMap<Self::Key, usize>,
+            key: &usize,
+            known_bindings: &Self::Map,
             (): &(),
-        ) -> BindingOptions<Self::Key, usize> {
+        ) -> BindingResult<Self, ()> {
             if *key == 0 {
                 // Key 0 maps to 0
-                BindingOptions::ValidBindings(vec![*key])
+                Ok(ValidBindings(vec![*key]))
```

Comment:
Note this function is
```
if *key==0 || known_bindings.get(&(key-1)).is_some() {
   Ok(ValidBindings(vec![*key]))
} else {
  Err(MissingIndexKeys(key![keys-1]))
}
```
(that is, the first two branches are the same). Plus comments which I have skipped.

Reply:
true, i've simplified this, thanks!